packages/graph-retriever/src/graph_retriever/edges/_base.py

```diff
@@ -34,6 +35,17 @@ class MetadataEdge(Edge):
         The value associated with the key for this edge
     """
 
+    def __init__(self, incoming_field: str, value: Any) -> None:
+        # `self.field = value` and `setattr(self, "field", value)` -- don't work
+        # because of frozen. we need to call `__setattr__` directly (as the
+        # default `__init__` would do) to initialize the fields of the frozen
+        # dataclass.
+        object.__setattr__(self, "incoming_field", incoming_field)
+
+        if isinstance(value, dict):
```

Comment:
Convert `dict` to `immutabledict` so that `MetadataEdge` is hashable (can be put in a set). Lot's of places rely on that and/or using the edge as a key.